### PR TITLE
Update Code Coverage Calculations

### PR DESCRIPTION
**Pull request by Javagedes**
Update code coverage calculations to have the ability to reorganize the report by INF using `stuart_parse` and `stuart_report.`

I have confirmed this as working using MuDevOpsWrapper for MU_BASECORE PRs:

Windows: [Pipelines - Run 20231102.11 (azure.com)](https://dev.azure.com/projectmu/mu/_build/results?buildId=57362&view=codecoverage-tab)
Linux: [Pipelines - Run 20231102.11 (azure.com)](https://dev.azure.com/projectmu/mu/_build/results?buildId=57363&view=codecoverage-tab)

### CODE REVIEW

**Review by cfernald — APPROVED** [review]

File: Jobs/PrGate.yml [hunk 1]
```diff
@@ -35,6 +35,9 @@ parameters:
   displayName: Perform Stuart PR Evaluation
   type: boolean
   default: true
+- name: calculate_code_coverage
```
Comment (makubacki) [inline Jobs/PrGate.yml:38]: Can you add the `type` to keep consistent with the existing parameter format?

File: Jobs/PrGate.yml [hunk 2]
```diff
@@ -164,19 +167,27 @@ jobs:
...
+        ${{ else }}:
+          extra_build_args: ''
 
-- job: PublishCoverage
-  dependsOn:
-  - ${{ each item in parameters.build_matrix }}:
-    - Build_${{ item.Key }}
-  condition: and(not(Canceled()), not(Failed()))
+- ${{ if eq(parameters.calculate_code_coverage, true) }}:
+  - job: PublishCoverage
```
Comment (makubacki) [inline Jobs/PrGate.yml:176]: Not related to the PR, but I filed an issue to try to optimize code coverage publishing in the pipeline here - https://github.com/microsoft/mu_devops/issues/267. Please let me know if you have some time to look into it.

File: Steps/PublishCodeCoverage.yml [hunk 3]
```diff
@@ -10,13 +10,31 @@ parameters:
   displayName: Perform self checkout step
   type: boolean
   default: true
+- name: setup_python
+  displayName: Setup Python
+  type: boolean
+  default: true
+- name: reorganize_by_inf
```
Comment (makubacki) [inline Steps/PublishCodeCoverage.yml:17]: Like the other file, it would be nice to keep the `type` on each parameter explicit for consistency.

**Review by makubacki — APPROVED** [review]

File: Steps/PublishCodeCoverage.yml (hunk 3 above)
Comment (Javagedes) [reply Steps/PublishCodeCoverage.yml:17]: Added typing to reorganize_by_inf and build_file parameters as requested. Thanks!

File: Jobs/PrGate.yml (hunk 1 above)
Comment (Javagedes) [reply Jobs/PrGate.yml:38]: added a type to calculate_code_coverage as requested! Thanks

File: Jobs/PrGate.yml (hunk 2 above)
Comment (Javagedes) [reply Jobs/PrGate.yml:177]: I've left a comment on the issue. I'd be happy to look more into it also. 

I'd like to get this in first, though, if that is acceptable.

Comment (makubacki) [reply Jobs/PrGate.yml:177]: Yes, that's an existing problem that doesn't need to block this.

**Pull request merged**